### PR TITLE
Update TypedServer demo

### PR DESCRIPTION
**Pull request by erszcz**
This PR updates the TypedServer demo to use current Gradient (instead of GradualizerEx) and makes sure it's runnable as is without any glitches.

The demo as of this PR still relies on two non-standard features in upstream Gradualizer:
- https://github.com/josefs/Gradualizer/pull/391
- https://github.com/josefs/Gradualizer/pull/392

The aim is to get rid of any customisations and either make the demo conform to the upstream feature set or merge the necessary changes upstream.

### CODE REVIEW

**Comment by erszcz** [comment]
@zuiderkwast This is the example code from the Friday live demo. To make it work the way it does (looking the from the utility / UX perspective) https://github.com/josefs/Gradualizer/pull/391 is necessary upstream. I didn't see any breakage in Gradualizer from incorporating that change, but OTOH I'm not yet done with all the relevant literature on the subject, which could highlight potential issues. Just letting you know what my thinking currently is.

File: examples/typed_gen_server/lib/typed_gen_server/stage1.ex [hunk 1]
```diff
@@ -41,27 +43,25 @@ defmodule TypedGenServer.Stage1.Server do
     end
   end
 
-  @impl true
+  # @impl true
```
Comment (Premwoik) [inline examples/typed_gen_server/lib/typed_gen_server/stage1.ex:46]: Why comment out this here and in the other places?

**Review by Premwoik — APPROVED** [review]
Looks good! Well done examples :D Thanks for adding the formatter of the `nonexhaustive` error :)

File: examples/typed_gen_server/lib/typed_gen_server/stage1.ex (hunk 1 above)
Comment (erszcz) [reply examples/typed_gen_server/lib/typed_gen_server/stage1.ex:46]: This only makes sense if `use GenServer` is also enabled at the top, but this in turn leads to a type error reported within `GenServer` implementation. I don't want this to clutter the output when demoing, but ultimately we would want these uncommented and the error sorted out.

Good point, though, I'll create a ticket to track this.

Comment (erszcz) [reply examples/typed_gen_server/lib/typed_gen_server/stage1.ex:46]: Tracked as https://github.com/esl/gradient/issues/64

**Comment by zuiderkwast** [comment]
> @zuiderkwast This is the example code from the Friday live demo. To make it work the way it does (looking the from the utility / UX perspective) [josefs/Gradualizer#391](https://github.com/josefs/Gradualizer/pull/391) is necessary upstream.

Thanks for inviting me to the presentation! It was very good! About the PR, I'm convinced now. :)

**Pull request merged**